web/src/stream_popover.js

```diff
@@ -565,9 +569,55 @@ export async function build_move_topic_to_stream_popover(
         event.stopPropagation();
     }
 
+    function update_warning_banner() {
+        // The warning banner only shows when
+        // 1) submit button is NOT disabled, and
+        // 2) target topic is an existing topic, and
+        // 3) propagate_mode is NOT change_one
+
+        // submit button
+        if ($("#move_topic_modal .dialog_submit_button")[0].disabled) {
+            $("#move_topic_modal .move_topic_form_warning_container").html("");
+            return;
+        }
+
+        // target topic
+        if (stream_widget_value !== undefined) {
+            const {new_topic_name} = get_params_from_form();
+            if (
+                !stream_topic_history.stream_has_topic(
+                    Number.parseInt(stream_widget_value, 10),
+                    new_topic_name,
+                )
+            ) {
+                $("#move_topic_modal .move_topic_form_warning_container").html("");
+                return;
+            }
+        }
```

Comment:
I wonder if we can simplify this nested if statement into two statements on the same level.